syntax = "proto3";

message secret {
   repeated int32 dataIds = 1;   // data id
   repeated string dataCIds = 2;  // data's cluster id
   repeated string cIds = 3; // cluster id
   repeated bytes cipherTexts = 4;
   repeated int32 exponents = 5;

}

message globalDist {
  repeated int32 dataIds = 1;
  repeated bytes cipherText = 2;
  repeated int32 exponent = 3;
}

service safeTransmission {
  rpc exchangeData (secret) returns (globalDist);
}